broker/tasks/iam.py

```diff
@@ -135,18 +135,38 @@ def delete_previous_server_certificate(operation_id: str, **kwargs):
     else:
         iam = iam_govcloud
 
-    for certificate in Certificate.query.filter(
+    certificates_to_delete = Certificate.query.filter(
         and_(
             Certificate.service_instance_id == service_instance.id,
             Certificate.id != service_instance.current_certificate_id,
         )
-    ).all():
+    ).all()
+    for certificate in certificates_to_delete:
         try:
             iam.delete_server_certificate(
                 ServerCertificateName=certificate.iam_server_certificate_name
             )
-        except iam_commercial.exceptions.NoSuchEntityException:
-            pass
+        except ClientError as e:
```

Comment:
The nested try/catch here is pretty ugly, but I'm not sure how else to handle it

Reply:
I think it might be tidier:

```
try:
    get_response = iam.get_server_certificate(...)
except ClientError as e:
    if "NoSuchEntity" in e.response["Error"]["Code"]:
        return # the cert is gone, we've got nothing to do
# now we know we can see the cert, so any errors should be treated as unexpected
iam.delete_server_certificate(...)
```

n.b. I put a `return` here - that assumes pulling this logic out into a function (which I think is probably a good idea). You could use `next` instead if you wanted to leave the loop